rust/common/mod.rs

```diff
@@ -51,6 +65,41 @@ impl Spannable for &str {
             LineColumn { line: end_line as u32, column: end_col as u32 },
         ))
     }
+
+    fn extract_annotated_line_col(
+        &self,
+        line: usize,
+        col: usize,
+        lines_before: usize,
+        lines_after: usize,
+    ) -> Option<String> {
+        let mut annotated = false;
+        let lines: Vec<_> = self
+            .lines()
+            .enumerate()
+            .map(|(i, line_string)| {
+                if i == line {
+                    annotated = true;
+                    format!(
+                        "{SYNTAX_ANNOTATED_INDICATOR_LINE}{line_string}\n{}{SYNTAX_ANNOTATED_INDICATOR_COL}",
+                        " ".repeat(SYNTAX_ANNOTATED_INDENT + col)
+                    )
```

Comment:
We add both the line indicator, and a following like with an `^` column indicator